### PR TITLE
Adaptive gradient descent for point_type=matrix

### PR DESCRIPTION
**Pull request by nguigs**
Update adaptive gradient descent to work with matrices

### CODE REVIEW

**Comment by nguigs** [comment]
@xpennec

File: tests/test_frechet_mean.py [hunk 1]
```diff
@@ -149,6 +149,18 @@ def test_estimate_and_belongs_default_gradient_descent_so_matrix(self):
         expected = True
         self.assertAllClose(result, expected)
 
+    @geomstats.tests.np_and_tf_only
```
Comment (ninamiolane) [inline tests/test_frechet_mean.py:152]: What is the problem with pytorch?

**Review by ninamiolane — APPROVED** [review]
Very nice!

**Comment by ninamiolane** [comment]
@nguigs what was the problem with tests in pytorch on this one?

File: tests/test_frechet_mean.py (hunk 1 above)
Comment (nguigs) [reply tests/test_frechet_mean.py:152]: This test is new, it cannot be used with pytorch because the matrix log is used but not implemented in pytorch
